lib/plugins/package/lib/zipService.js

```diff
@@ -37,17 +32,17 @@ module.exports = {
     if (excludeDevDependencies) {
       this.serverless.cli.log('Excluding development dependencies...');
 
-      return BbPromise.bind(this)
-        .then(() => excludeNodeDevDependenciesMemoized(servicePath))
-        .then(exAndInNode => {
-          params.exclude = _.union(params.exclude, exAndInNode.exclude); //eslint-disable-line
-          params.include = _.union(params.include, exAndInNode.include); //eslint-disable-line
-          return params;
-        })
-        .catch(() => params);
+      try {
+        const exAndInNode = await excludeNodeDevDependenciesMemoized(servicePath);
+        params.exclude = _.union(params.exclude, exAndInNode.exclude); //eslint-disable-line
+        params.include = _.union(params.include, exAndInNode.include); //eslint-disable-line
+        return params;
+      } catch (e) {
+        return params;
+      }
     }
 
-    return BbPromise.resolve(params);
+    return params;
```

Comment:
If I see correctly, a more proper setup of this logic would be as follows:

```javascript
const exAndInNode = await (async () => {
  try { return await excludeNodeDevDependenciesMemoized(servicePath); } catch {}
})();
if (exAndInNode) {
  params.exclude = _.union(params.exclude, exAndInNode.exclude);
  params.include = _.union(params.include, exAndInNode.include); 
}
return params;
```

Reply:
The inline `await` seems a bit harder to read to me (subjective). What about:
```js
      let exAndInNode;
      try {
        exAndInNode = await excludeNodeDevDependenciesMemoized(servicePath);
      } catch {}
      if (exAndInNode) {
        params.exclude = _.union(params.exclude, exAndInNode.exclude); //eslint-disable-line
        params.include = _.union(params.include, exAndInNode.include); //eslint-disable-line
      }
      return params;
```

I can go with whichever you prefer

Reply:
It's totally fine! Technically I proposed IIFE simply to use `const` over `let`, but both ways are ok imo (it's purely matter of an opinion)